hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaMessagesProducingObserver.java

```diff
@@ -0,0 +1,47 @@
+package pl.allegro.tech.hermes.frontend.producer.kafka;
+
+import com.google.common.base.Preconditions;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResult;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingException;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingObserver;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResults;
+
+import java.util.Collection;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class KafkaMessagesProducingObserver implements BrokerMessagesProducingObserver {
+
+    private final Collection<BrokerMessagesProducingResult> validationResults;
+    private final CountDownLatch countDownLatch;
+    private final long timeoutMs;
+
+    public KafkaMessagesProducingObserver(int messagesCounter, long timeoutMs) {
+        Preconditions.checkArgument(messagesCounter > 0, "Must observe positive number of messages to be produced");
+        Preconditions.checkArgument(timeoutMs > 0L, "Timeout must be defined");
+        this.countDownLatch = new CountDownLatch(messagesCounter);
+        validationResults = new ArrayBlockingQueue<>(messagesCounter);
```

Comment:
Why you use queue here ? Is there any reason other than the fact it is thread safe ? Maybe simple synchronized list is enough ?